Pass along locale when formatting currencies.

muirrn opened the pull request:
The currency symbol depends on the locale (e.g. USD uses "USD" not "$" in en-AU).

camertron commented:
Hey @muirrn I'm seeing a bunch of failing tests and it doesn't look like your change worked?

muirrn commented:
I rebased without re-running tests. Let me take a look...

muirrn commented:
I rebased on the fix-parent-locales branch to fix my test (the current master CLDR info has a lot of problems because the "base" locale info in v29 of the CLDR is not handled at all by the current version of ruby-cldr gem).

I fixed other failures by defaulting the locale to ":en" properly.

camertron commented:
Ok, this is looking pretty good. Now that fix-parent-locales has been merged, would you mind merging/rebasing master again?

camertron commented:
Published in v4.2.0.
